documentation/docs/libraries/java/how_to/_05_build_output/_expected_output.md

```diff
@@ -1,15 +1,103 @@
 ```json
 {
-  "type": 3,
-  "amount": "1000000",
-  "unlockConditions": [
-    {
-      "type": 0,
-      "address": {
+  "jsonObject": {
```

Comment:
Why `jsonObject` ?

Reply:
Haha, I have no clue. I'm not a Java dev. This is just how it was printed with the `Gson` thingy afaik

Reply:
But it wasn't printed before?

Reply:
Yeah I'm actually wondering about that too. Let me check

Reply:
I have no clue who prettified the previous output. But if I just use `.toString()` it won't be pretty printed

Reply:
Lucas did it probably manually
The key probably gets added because the Output object extends AbstractObject which has a field named like this
```
public class AbstractObject {

    private JsonObject jsonObject;
```

Reply:
We need to get rid of this

Reply:
Will look at an alternative for Gson than I guess

Reply:
@samuel-rufi or @kwek20 can you advise something here please?

Reply:
Solved

Reply:
My Java knowledge is very bad but I wouldn't say this has been solved. You changed the way it's printed in the example but someone else trying to print it would still have that `jsonObject` stuff. We need to have a way to properly and TIP-compliantly serialise to JSON.

Reply:
The toString() implementation doesn't print it, but there it's not pretty printed

Reply:
Hmm I see. Such an annoying language.

Reply:
Exactly this is just about pretty printing. I don't care about the TIP there at all ;)

Reply:
And why do we even care how people print it?

Reply:
Why would we not care? It doesn't make any sense that the content is not the same pretty printed or not. Only the format should differ. This is very confusing.

Reply:
I still think our official outputs should be TIP-compliant.

Reply:
But that can be another PR ofc

Reply:
> I still think our official outputs should be TIP-compliant.

Sorry that wasn't clear, I just mean it would be nice to have a concise way to pretty print. But maybe that's just Java being overly ugly idk.

Reply:
I mean it's up to the user if he wants to pretty print this things. But obviously it would be nice to recommend the best implementation in our examples. You are right.
Not sure if there is a better solution in Java. If we have a really strong opinion on that we could also add the "pretty printer" to the Java binding 🤔

Reply:
Ofc it's their choice, what I'm saying is that the possibility of creating a non TIP-compliant JSOn should not exist at all. We don't know what people are going to do with the output, if they try to re-parse it, it'll fail. Our library shouldn't serialise something it can't deserialise. But I agree that it's out of scope of this PR. Happy to merge.

Reply:
@Dr-Electron can you open an issue about that pls?